app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListFragment.kt

```diff
@@ -66,111 +74,150 @@ class ConnectionsListFragment : BaseFragment(), ConnectionsListContract.View,
+            it.getContentIfNotHandled()?.let { supportEmail ->
+                activity?.startMailApp(supportEmail)
+            }
+        })
+        viewModel.updateListItemEvent.observe(this, Observer<ViewModelEvent<Int>> {
+            it.getContentIfNotHandled()?.let { itemIndex ->
+                viewModel.listItemsValues.getOrNull(itemIndex)?.let { item ->
+                    adapter.updateListItemName(item.guid, item.name)
+                }
+            }
+        })
+    }
+
+    private fun setupViews() {
+        connectionsListView?.layoutManager = LinearLayoutManager(activity ?: return)
+        connectionsListView?.adapter = adapter
+        emptyView?.setOnClickListener(this)
+        headerDecorator = SpaceItemDecoration(
+            context = activity ?: return
+        ).apply {
+            connectionsListView?.addItemDecoration(this)
         }
-        val viewIsEmpty = adapter.isEmpty
-        emptyView?.setVisible(viewIsEmpty)
-        connectionsListView?.setVisible(!viewIsEmpty)
-    }
-
-    override fun showApiErrorView(message: String) {
-        activity?.showWarningDialog(message = message)
-    }
-
-    override fun showSupportView(supportEmail: String?) {
-        activity?.startMailApp(supportEmail)
-    }
-
-    override fun showConnectView(connectionGuid: GUID) {
-        activity?.addFragment(ConnectProviderFragment.newInstance(connectionGuid = connectionGuid))
+        binding.executePendingBindings()
     }
 
-    override fun showConnectionNameEditView(
-        connectionGuid: String,
-        connectionName: String,
-        requestCode: Int
-    ) {
-        val dialog = EditConnectionNameDialog.newInstance(connectionGuid, connectionName).also {
-            it.setTargetFragment(this, requestCode)
+    private fun showPopupMenu(view: View?, itemIndex:  Int) {
+        val layoutInflater = context?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+        val popupView = layoutInflater.inflate(R.layout.popup_menu_layout, null)
+        val popupWindow = PopupWindow(
+            popupView,
+            ViewGroup.LayoutParams.WRAP_CONTENT,
+            ViewGroup.LayoutParams.WRAP_CONTENT
+        )
+        popupWindow.isOutsideTouchable = true
+        popupWindow.isFocusable = true
+        popupWindow.showAsDropDown(view)
+
+        val renameView = popupView.findViewById<LinearLayout>(R.id.renameView)
+        val reconnectView = popupView.findViewById<LinearLayout>(R.id.reconnectView)
+        val contactSupportView = popupView.findViewById<LinearLayout>(R.id.contactSupportView)
+        val deleteView = popupView.findViewById<LinearLayout>(R.id.deleteView)
+        val deleteImageView = popupView.findViewById<ImageView>(R.id.deleteImageView)
+        val deleteTextView = popupView.findViewById<TextView>(R.id.deleteTextView)
+
+        viewModel.checkStatusOfConnection(itemIndex)
+
+        viewModel.reconnectViewVisibility.observe(this, Observer<Int> {
```

Comment:
Observer (and MutableLiveData) is redundant because value not changing while popupView is alive (on top)